### PR TITLE
Remove cc option nocheck

### PR DESCRIPTION
**Pull request by SebastianBoe**
zephyr_cc_option_nocheck() is the same as zephyr_compile_definitions() so we don't need to introduce zephyr_cc_option_nocheck().

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -195,7 +195,7 @@ zephyr_cc_option(-fno-strict-overflow)
 zephyr_cc_option(-Wno-pointer-sign)
 
 if(CONFIG_STACK_CANARIES)
-  zephyr_cc_option_nocheck(-fstack-protector-all)
+  zephyr_compile_options(-fstack-protector-all)
```
Comment (nashif) [inline CMakeLists.txt]: maybe use zephyr_compile_options_ifdef instead?

Comment (SebastianBoe) [reply CMakeLists.txt]: Done.

**Comment by nashif** [comment]
btw, the naming of the two options is unlucky, zephyr_compile_options vs zephyr_cc_options, this will be confused and there is no indication of what the difference is by looking at the name only.

**Comment by SebastianBoe** [comment]
I agree, I remember thinking of using a different name, but eventually settled on cc_option to stay close
to KBuild's equivalent cc-option. https://www.kernel.org/doc/Documentation/kbuild/makefiles.txt.

**Review by nashif — APPROVED** [review]

**Pull request merged**